src/Simplify.elm

```diff
@@ -2922,6 +2929,33 @@ reportEmptyListFirstArgument ( ( moduleName, name ), function ) =
 -- STRING
 
 
+stringFromListChecks : CheckInfo -> List (Error {})
+stringFromListChecks checkInfo =
+    case Node.value checkInfo.firstArg of
+        Expression.ListExpr [] ->
+            [ Rule.errorWithFix
+                { message = "Calling String.fromList [] will result in " ++ emptyStringAsString
+                , details = [ "You can replace this call by " ++ emptyStringAsString ++ "." ]
+                }
+                checkInfo.fnRange
+                [ Fix.replaceRangeBy checkInfo.parentRange emptyStringAsString ]
+            ]
+
+        Expression.ListExpr ((Node keep _) :: []) ->
```

Comment:
One thing that we could do later on, is the notice calls like `String.fromList (List.singleton x)` and consider those to be the same as `String.fromList [ x ]`. But that's for later (and for a lot more things)

Reply:
absolutely.